ch_tools/chadmin/cli/data_store_group.py

```diff
@@ -405,14 +379,42 @@ def find_paths_to_part_with_lost_keys(ctx: Context, root_path: str) -> List[str]
                 s3_client, disk_conf.bucket_name, full_key
             ):
                 logging.debug("Not found key {}", full_key)
-                result.append(path)
+                parts_paths_with_lost_keys.append(path)
+
+                partition_list = get_partitions_by_path(ctx, [path])
+                try_restore_data(ctx, partition_list, reattach, detach)
+
                 logging.debug(
-                    "Add part to check with path", result[-1][0], result[-1][1]
+                    "Add part to check with path",
+                    parts_paths_with_lost_keys[-1][0],
+                    parts_paths_with_lost_keys[-1][1],
                 )
                 break
 
-    logging.debug("Found parts with missing s3 keys. Local paths of parts: {}", result)
-    return result
+    partition_list = get_partitions_by_path(ctx, parts_paths_with_lost_keys)
+    print_response(ctx, partition_list, default_format="table")
+
+    logging.debug(
+        "Found parts with missing s3 keys. Local paths of parts: {}",
+        parts_paths_with_lost_keys,
+    )
+
+
+def try_restore_data(ctx, partition_list, reattach, detach):
+    if not (reattach and detach):
```

Comment:
Check covers only `reattach=true ; detach=true`. Lest remove it because we already have checked this as cli opt. Or make it `not (reattach xor detach)`.